pkg/utils/utils.go

```diff
@@ -125,3 +125,27 @@ func (r *ResolutionMetadata) Fork() *ResolutionMetadata {
 func LogDBStats(ctx context.Context, log logger.Logger, method string, reads uint32, writes uint32) {
 	log.InfoWithContext(ctx, "db_stats", logger.String("method", method), logger.Uint32("reads", reads), logger.Uint32("writes", writes))
 }
+
+type ConcurrentSet struct {
+	M map[string]bool
+	sync.RWMutex
+}
+
+func NewConcurrentSet() *ConcurrentSet {
+	return &ConcurrentSet{
+		M: make(map[string]bool),
```

Comment:
We should try not to use this for ListObjects because it creates contention between goroutines unnecessarily.  It creates read/write lock contention between them when we don't need to.

Reply:
Removed